lib/stock_markets/file_data_processor.rb

```diff
@@ -1,32 +1,61 @@
-require 'json'
+# frozen_string_literal: true
 require 'csv'
 
 module StockMarkets
   class FileDataProcessor
+    LAST_UPDATE_MARKET_MINIMAL_YEARS_COUNT = 2
+    ACTIVE_MARKET_STATUS_NAME = 'ACTIVE'
+    MARKET_NAME_CSV_COLUMN = :nameinstitution_description
 
-    attr_accessor :format, :data
+    attr_accessor :data
 
     def initialize
-      @format = 'json'
       @data = []
+      @parsed_csv = nil
     end
 
-    def load_from_disk
-      self.data = CSV.table(File.expand_path("../../data/stock_markets.csv", __FILE__))
+    def load_from_disk!
+      self.parsed_csv = CSV.table(StockMarkets.configuration.data_file_path)
       self
     end
 
-    def transform_to_hash
-      raise TypeError, 'Invalid object of CSV provided!' unless data.instance_of?(CSV::Table)
+    def transform_to_hash(key_name: :mic)
+      raise TypeError, 'Invalid object of CSV provided!' unless parsed_csv.instance_of?(CSV::Table)
 
-      self.data = data.map do |data_row| 
-        [data_row[:mic] || data_row[:name], data_row.to_h]
-      end.to_h
+      self.data = parsed_csv.each_with_object({}) do |data_row, result_hash|
+        if market_recently_updated_proc.call(data_row, key_name)
+          result_hash[data_row[key_name.to_sym]] =  data_row.to_h
+        end
+      end
       self
     end
 
-    def load_data!
-      load_from_disk.transform_to_hash.data
+    def load_data_for_mics!
+      transform_to_hash.data
+    end
+
+    def load_data_for_names!
+      transform_to_hash(key_name: MARKET_NAME_CSV_COLUMN).data
+    end
+
+    private
+
+    attr_accessor :parsed_csv
+
+    def last_update_market_minimal_date
+      return @_last_update_market_minimal_date if @_last_update_market_minimal_date
```

Comment:
dont you like 

```ruby
@_last_update_market_minimal_date ||= begin 
  ....
end
```
? :)

Reply:
@AlexKlim https://github.com/rubocop-hq/ruby-style-guide#no-BEGIN-blocks I think, it is not good idea :(

Reply:
I need more details :)

Reply:
just avoid of using id :D